dir.targets

```diff
@@ -1,154 +1,40 @@
-            if (client.Proxy != null) client.Proxy.Credentials = System.Net.CredentialCache.DefaultCredentials;
-            var tryCount = 1;
-            var maxTries = 3;
-
-            while (tryCount <= maxTries)
-            {
-                try
-                {
-                    Log.LogMessage("Attempting to download {0}...", Address);
-            client.DownloadFile(Address, tempFile);
-                    break;
-                }
-                catch (System.Net.WebException e)
-                {
-                    tryCount++;
-                    if (tryCount > maxTries)
-                    {
-                        throw;
-                    }
-                    else
-                    {
-                        Log.LogMessage(MessageImportance.High, "Download failed, retrying: {0}", e.Message);
-                    }
-                }
-            }
-
-            try
-            {
-                if (!File.Exists(filename))
-                   File.Move(tempFile, filename);
-            }
-            finally
-            {
-              if (File.Exists(tempFile))
-                  File.Delete(tempFile);
-            }
-        ]]>
-      </Code>
-    </Task>
-  </UsingTask>
-
-  <!--
-    Use a semaphore file to determine the need to restore build tools to avoid conflicts with locked binaries.
-  -->
-  <PropertyGroup>
-    <BuildToolsSemaphore>$(ToolsDir)BuildTools.semaphore</BuildToolsSemaphore>
-  </PropertyGroup>
-
   <!--
     Needed to avoid the IntialTargets from having an Output which ends up getting
     added to the output references when you have a project to project reference.
   -->
-  <Target Name="_RestoreBuildToolsWrapper" DependsOnTargets="_DisplayBuildInfo;_RestoreBuildTools" />
+  <Target Name="_RestoreBuildToolsWrapper" DependsOnTargets="_DisplayBuildInfo;_CopyCompilers" />
 
   <Target Name="_DisplayBuildInfo">
     <Message Importance="High" Text="Build binary target directory: '$(BaseOutputPathWithConfig)'" />
   </Target>
 
-  <Target Name="_CopyCompilers" AfterTargets="_RestoreBuildTools" Condition="'$(UseRoslynCompilers)' == 'true'">
-    <ItemGroup>
-      <BuildFiles Include="$(PackagesDir)Microsoft.Net.Compilers*\build\*" />
-      <ToolsFiles Include="$(PackagesDir)Microsoft.Net.Compilers*\tools\*" />
-    </ItemGroup>
+  <Target Name="_CopyDataSetPermissions" AfterTargets="Build; Rebuild" Condition="'$(IsTestProject)' != 'true'"/>
 
-    <Copy
-      SourceFiles="@(BuildFiles);@(ToolsFiles)"
-      DestinationFolder="$(OutputPath)tools"
-      SkipUnchangedFiles="true"
-    />
-    <Exec Condition="'$(OS)' != 'Windows_NT'" Command="find '$(OutputPath)tools' -name &quot;*.exe&quot; -exec chmod &quot;+x&quot; '{}' ';'" />
-  </Target>
-
-  <Target Name="_CopyCompilers" AfterTargets="_RestoreBuildTools" Condition="'$(UseRoslynCompilers)' == 'true'">
+  <Target Name="_CopyCompilers"
+          BeforeTargets="_CopyDataSetPermissions"
```

Comment:
What does this do?  Can you structure it as a DependsOnTargets in the other target instead if it's still necessary?